compiler/noirc_errors/src/position.rs

```diff
@@ -51,27 +81,49 @@ impl<T> std::borrow::Borrow<T> for Spanned<T> {
     }
 }
 
+#[derive(
+    Copy, Clone, Serialize, Eq, PartialEq, Ord, PartialOrd, Debug, Deserialize, Hash, Default,
+)]
+pub struct SrcId(usize);
+
+impl From<usize> for SrcId {
+    fn from(value: usize) -> Self {
+        Self(value)
+    }
+}
+
+impl From<SrcId> for usize {
+    fn from(val: SrcId) -> Self {
+        val.0
+    }
+}
+
 #[derive(
     PartialEq, PartialOrd, Eq, Ord, Hash, Debug, Copy, Clone, Default, Deserialize, Serialize,
 )]
-pub struct Span(ByteSpan);
+pub struct Span(ByteSpan, SrcId);
```

Comment:
If a `SrcId` is meant to represent a File, do we need `Location` at all anymore?

I've yet to see this whole PR yet so I don't know how a SrcId would correspond to a FileId yet, e.g. if they're convertible or not.

Reply:
Location removed